tests/merge-discovery-resilience.test.sh

```
@@ -2724,6 +2747,106 @@ assert_eq "$([ -s "$B28_AUDIT" ] && echo nonempty || echo empty)" "empty" \
+_run_lib_call "success-mixed-base" \
+  'discover_open_prs' \
+  "export UBERDEV_AUDIT_LOG_PATH='$BC_AUDIT'; export FAKE_GH_CALL_LOG='$BC_CALLLOG'"
+assert_eq "$_LB_EXIT" "0" "BC2a: exit 0 (discover_open_prs never aborts the caller)"
+BC_NUMS="$(jq -c '[.[].number]' <<<"${_LB_STDOUT:-[]}" 2>/dev/null)"
+assert_eq "${BC_NUMS:-PARSE_FAILED}" "[201,202,203,204]" \
+  "BC2b: all 4 open non-draft PRs come back regardless of base; the draft (205) does not"
+# Paired anti-vacuity: prove the rooted filter WOULD have dropped rows from
+# this very input, so BC2b is a real "un-rooted" result and not a fixture that
+# happens to be rooted already.
+BC_ROOTED="$(
+  PATH="$FAKE_GH_DIR:$PATH" bash -c ". '$LIB' && _uberdev_discover_stack_filter main '$(FAKE_GH_MODE=success-mixed-base "$FAKE_GH_DIR/gh" pr list --jq '[.[] | select(.isDraft==false)]' 2>/dev/null)'" 2>/dev/null
```

Shellcheck (0.11.0)

[info] PATH was modified in a subshell. That change might be lost.

(SC2031)